launchable/commands/subset.py

```diff
@@ -263,18 +272,36 @@ def subset(
     session_id = None
     tracking_client = TrackingClient(Tracking.Command.SUBSET, app=app)
     try:
-        session_id = find_or_create_session(
-            context=context,
-            session=session,
-            build_name=build_name,
-            flavor=flavor,
-            is_observation=is_observation,
-            links=links,
-            is_no_build=is_no_build,
-            lineage=lineage,
-            tracking_client=tracking_client,
-            test_suite=test_suite,
+        if session_name:
+            if not build_name:
+                raise click.UsageError(
+                    '--build option is required when you use a --session-name option ')
+            sub_path = "builds/{}/test_session_names/{}".format(build_name, session_name)
+            client = LaunchableClient(test_runner=context.invoked_subcommand, app=context.obj, tracking_client=tracking_client)
+            res = client.request("get", sub_path)
+            res.raise_for_status()
+            session_id = "builds/{}/test_sessions/{}".format(build_name, res.json().get("id"))
+        else:
+            session_id = find_or_create_session(
+                context=context,
+                session=session,
+                build_name=build_name,
+                flavor=flavor,
+                is_observation=is_observation,
+                links=links,
+                is_no_build=is_no_build,
+                lineage=lineage,
+                tracking_client=tracking_client,
+                test_suite=test_suite,
+            )
+    except click.UsageError as e:
+        click.echo(
```

Comment:
Also, I'd like to capture this exception in the telemetry.

Reply:
https://github.com/launchableinc/cli/pull/1001/commits/d64326a470051c63b90777220acdcda8a8a3a1a7